packages/components/src/shared/src/useFontFaceReady.ts

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useState } from "react";
+
+const useFontFaceReady = () => {
```

Comment:
I beleive the current implementation could result in error like this one
![image](https://user-images.githubusercontent.com/38871812/140548663-61e04760-599d-4ea7-9134-feca2a363e19.png)

This is due to the fact that the hook could be unmounted before the callback has been resolved, resulting in calling setReady on an unmounted hook

I believe this is a safer approach to this hook

```
const useFontFaceReady = () => {
    const [ready, setReady] = useState(false);

    useEffect(() => {
        let isCancelled = false;
        
        const loadFonts = async () => {
            await document.fonts.ready;
            if (!isCancelled) {
                setReady(true);
            }
        };

        loadFonts();

        return () => {
            isCancelled = true;
        };
    }, []);

    return ready;
};
```